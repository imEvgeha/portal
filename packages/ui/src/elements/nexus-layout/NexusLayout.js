import React from 'react';
import {IfEmbedded} from '@vubiquity-nexus/portal-utils/lib/Common';
import DOP from '@vubiquity-nexus/portal-utils/lib/DOP';
import {Outlet} from 'react-router-dom';
import NexusNavigation from '../nexus-navigation/NexusNavigation';
import './NexusLayout.scss';

const NexusLayout = ({children}) => (
    <div className="nexus-c-app-layout">
        <IfEmbedded>
            <DOP />
        </IfEmbedded>
        <IfEmbedded value={false}>
            <NexusNavigation />
        </IfEmbedded>
        <div className="nexus-c-app-layout__main">
            <Outlet />
        </div>
    </div>
);

export default NexusLayout;
