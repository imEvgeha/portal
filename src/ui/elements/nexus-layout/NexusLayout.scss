@import '../../../styles/config.scss';

.nexus-c-app-layout {
    display: flex;
    margin: 0 auto;
    flex-direction: row;
    min-height: 100%;
    width: 100%;
    height: 100%;

    .nexus-navbar-user{
        .dropdown-menu{
            right: 0;
            left: auto;
        }
    }
}

.nexus-c-app-layout__main {
    width: 100%;
    padding: 0;
    background-color: map-get($colors, primary);
    height: 100%;
    overflow-y: auto;
}
