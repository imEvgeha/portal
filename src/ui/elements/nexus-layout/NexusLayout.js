import React from 'react';
import './NexusLayout.scss';
import {IfEmbedded} from '../../../util/Common';
import DOP from '../../../util/DOP';
import NexusNavigation from '../nexus-navigation/NexusNavigation';

const NexusLayout = ({children}) => (
    <div className="nexus-c-app-layout">
        <IfEmbedded>
            <DOP />
        </IfEmbedded>
        <IfEmbedded value={false}>
            <NexusNavigation />
        </IfEmbedded>
        <div className="nexus-c-app-layout__main">
            {children}
        </div>
    </div>
);

export default NexusLayout;
