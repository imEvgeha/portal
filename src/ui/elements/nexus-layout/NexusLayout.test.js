import React from 'react';
import {shallow} from 'enzyme';

import NexusLayout from './NexusLayout';

describe('NexusLayout', () => {
    let wrapper;
    beforeAll(() => {
        wrapper = shallow(<NexusLayout />);
    });
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

