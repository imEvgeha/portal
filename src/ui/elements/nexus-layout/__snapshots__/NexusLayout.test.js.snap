// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusLayout should match snapshot 1`] = `
<div
  className="nexus-c-app-layout"
>
  <IfEmbedded
    value={true}
  >
    <DOP />
  </IfEmbedded>
  <IfEmbedded
    value={false}
  >
    <withRouter(Connect(NexusNavigation)) />
  </IfEmbedded>
  <div
    className="nexus-c-app-layout__main"
  >
    <withRouter(Connect(NexusBreadcrumb)) />
  </div>
</div>
`;
